src/main/java/com/codessquad/qna/web/service/QuestionService.java

```diff
@@ -20,11 +22,35 @@ public void save(Question question) {
         questionRepository.save(question);
     }
 
+    public void save(User writer, String title, String contents) {
+        Question question = new Question(writer, title, contents);
+        save(question);
+    }
+
+    public void update(Question question, String title, String contents) {
+        question.update(title, contents);
+        save(question);
+    }
+
+    public void delete(Question question) {
+        questionRepository.delete(question);
+    }
+
     public List<Question> findAll() {
         return questionRepository.findAll();
     }
 
-    public Optional<Question> findById(Long id) {
-        return questionRepository.findById(id);
+    public Question findById(Long id) {
+        Question question = questionRepository.findById(id)
+                .orElseThrow(() -> new QuestionNotFoundException());
```

Comment:
```suggestion
                .orElseThrow(QuestionNotFoundException::new);
```
자바의 람다는 메서드 오퍼레이터를 지원합니다.

Reply:
넵 공부해보겠습니다!